src/parser/jobs/brd/PitchPerfect.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * @author Yumiya
+ */
+import Module from 'parser/core/Module'
+import ACTIONS, {getAction} from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import math from 'mathjsCustom'
+import {matchClosest} from 'utilities'
+
+// Relevant crit buffs
+const CRIT_MODIFIERS = [
+	{
+		id: STATUSES.BATTLE_LITANY.id,
+		strenght: 0.15,
+	},
+	{
+		id: STATUSES.CHAIN_STRATAGEM.id,
+		strenght: 0.15,
+	},
+	{
+		id: STATUSES.CRITICAL_UP.id,
+		strenght: 0.02,
+	},
+	{
+		id: STATUSES.THE_SPEAR.id,
+		//fuck royal road
+		strenght: 0.10,
+	},
+	{
+		id: STATUSES.STRAIGHT_SHOT.id,
+		strenght: 0.10,
+	},
+]
+
+const DHIT_MOD = 1.25
+
+const DISEMBOWEL_STRENGTH = 0.05
+const TRAIT_STRENGTH = 0.20
+
+const DEVIATION_PRECISION = 3
+
+export default class PitchPerfect extends Module {
+	static handle = 'pitchPerfect'
+	static dependencies = [
+		'additionalEvents', // eslint-disable-line xivanalysis/no-unused-dependencies
+		'combatants',
+		// Relying on the normaliser from this for the hit type fields
+		'hitType', // eslint-disable-line xivanalysis/no-unused-dependencies
+	]
+
+	// Represents a map of IDs and statuses for each enemy in this parse
+	_enemies = {}
+
+	// Represents the player statuses
+	_player = {
+		statuses: {},
+		hasStatus(statusId) { return this.statuses[statusId] || false },
+	}
+
+	normalise(events) {
+
+		const damageInstances = []
+		const ppEvents = []
+
+		for (const event of events) {
+			// Registers buffs/debuffs statuses on the respective entity (either player or enemies)
+			if (event.type.startsWith('apply') || event.type.startsWith('remove')) {
```

Comment:
I'd almost rather checking this w/ something like `(apply|remove)(de)?buff` just so new event types down the road aren't going to fuck it.